src/platform/packages/private/kbn-scout-reporting/src/reporting/jest/reporter.ts

```diff
@@ -0,0 +1,256 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+import type { Config, AggregatedResult, TestContext } from '@jest/reporters';
+import { BaseReporter } from '@jest/reporters';
+import { TestResult } from '@jest/types';
+import { ToolingLog } from '@kbn/tooling-log';
+import {
+  type CodeOwnerArea,
+  CodeOwnersEntry,
+  findAreaForCodeOwner,
+  getCodeOwnersEntries,
+  getOwningTeamsForPath,
+} from '@kbn/code-owners';
+import { SCOUT_REPORT_OUTPUT_ROOT, ScoutJestMetadata } from '@kbn/scout-info';
+import path from 'node:path';
+import { REPO_ROOT } from '@kbn/repo-info';
+import { ScoutJestReporterOptions } from './options';
+import {
+  datasources,
+  generateTestRunId,
+  getTestIDForTitle,
+  ScoutEventsReport,
+  ScoutFileInfo,
+  ScoutReportEventAction,
+  type ScoutTestRunInfo,
+  uploadScoutReportEvents,
+} from '../../..';
+
+/**
+ * Scout Jest reporter
+ */
+export class ScoutJestReporter extends BaseReporter {
```

Comment:
We talked offline about this - it's good to keep in mind: I see we're extending `BaseReporter` which is great! I see Jest also provides a `CoverageReporter` (a subclass of `BaseReporter`) which could provide valuable coverage metrics (for free, but at the expense of longer test run?). Something worthwhile to investigate in the future...

Reply:
Defintely something to consider in a future PR, but it's out of scope in this one.